code/modules/modular_computers/file_system/programs/file_browser.dm

```diff
@@ -49,6 +49,17 @@
 				return
 			file.filename = newname
 			return TRUE
+		if("PRG_usbrenamefile")
+			if(!RHDD)
+				return
+			var/datum/computer_file/file = RHDD.find_file_by_name(params["name"])
+			if(!file)
+				return
+			var/newname = params["new_name"]
+			if(!newname)
+				return
+			file.filename = newname
```

Comment:
Are you sure you don't want any validation on this?

Reply:
Sure, I'll change it to use `/proc/reject_bad_name`